Update hackathon startup guide (#814)
near/docs: docs/develop/basics/hackathon-guide.md

#### Linux and macOS

1. Install `npm` and `node` using a package manager such as `nvm`. Sometimes there are issues
   using Ledger due to how macOS handles node packages related to USB devices.
   [[click here]](https://nodejs.org/en/download/package-manager/)
2. Ensure you have installed Node version 12 or above.
3. Install `near-cli` globally by running:

```bash
npm install -g near-cli
```

#### Windows

> **Note:** For Windows users, we recommend using Windows Subsystem for Linux (`WSL`).

1. Install `WSL` [[click here]](https://docs.microsoft.com/en-us/windows/wsl/install-manual#downloading-distros)
2. Install ` Node.js` [[click here]](https://nodejs.org/en/download/package-manager/)
3. Change `npm` default directory [[click here]](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally#manually-change-npms-default-directory)
   - This is to avoid any permission issues with `WSL`
4. Open `WSL` and install `near-cli` globally by running:

```bash
npm install -g near-cli
```

## First steps

1) Now that you have an account, test out a simple `testnet` app and interact with the blockchain.

> Try [Counter](https://examples.near.org/rust-counter) or [Guest Book](https://examples.near.org/guest-book).

Reviewer: ```suggestion
Try out [Guest Book](https://near-examples.github.io/guest-book/) and sign and send your first transaction on the blockchain.
```

I think linking to a deployed site where users can immediately sign and send transactions and then goto Explorer to see the transactions would be a better fit here.

2) Look around in [NEAR Explorer](https://explorer.testnet.near.org). Here you can search for all transactions
and blocks produced on NEAR. Try searching for the account you just created and see the transactions you've

> created with Counter or Guest Book. 

Reviewer: ```suggestion
created with Guest Book. 
```

3) Try running your first `near-cli` command: [`near login`](/docs/tools/near-cli#near-login). This will
redirect you to your NEAR Wallet and save your `testnet` account keys locally. _Look for them in a hidden file
under your HOME folder (`~/.near-credentials`)_

4) Try `create-near-app`! Run `npx create-near-app your-awesome-project` in your terminal. _(Note: this requires
[Node.js](https://nodejs.org/en/).)_ This is the easiest way to launch a fullstack app on the NEAR blockchain

> in under 5 minutes.

Reviewer: We should delete this line as we are moving away from `create-near-app`

Author: Removed 👍

> 5) Head to [NEAR Examples](https://near.dev) and test out some example applications. You can clone and play

Reviewer: ```suggestion
4) Head to [NEAR Examples](https://near.dev) and test out some example applications. You can clone and play
```

  
- We currently support developing smart contracts in
  - [Rust](https://www.rust-lang.org/)
  - [AssemblyScript](https://assemblyscript.org/introduction.html) (Note: AssemblyScript is not recommended for mission-critical contracts)

## Boilerplate apps

> TBD

Reviewer 2: maybe we can put some demo projects here as well?

Reviewer: totally! Working on updating near.dev as we speak which will have more open source example projects.

Author: I have added a table with links/descriptions to all the examples available at `near.dev`.

I can update it as soon as we have more projects available.
